test/server/TypeDBClusterRunner.java

```diff
@@ -55,33 +91,100 @@ protected String name() {
 
     @Override
     protected int port() {
-        return ports.first();
+        return ports.external();
+    }
+
+    public Ports ports() {
+        return ports;
+    }
+
+    public Set<Ports> peers() {
+        return peers;
+    }
+
+    public Map<String, String> remainingServerOpts() {
+        return remainingServerOpts;
     }
 
     @Override
-    public void start() {
-        verifyPortUnused(ports.second());
-        verifyPortUnused(ports.third());
-        super.start();
+    protected void verifyPortUnused() {
+        super.verifyPortUnused(port());
+        super.verifyPortUnused(ports().internalZMQ());
+        super.verifyPortUnused(ports().internalGRPC());
     }
 
     @Override
     protected List<String> command() {
+        Map<String, String> serverOpts = new HashMap<>();
+        serverOpts.putAll(portOptions(ports));
+        serverOpts.putAll(peerOptions(peers));
+        serverOpts.putAll(remainingServerOpts);
+
         List<String> command = new ArrayList<>();
         command.addAll(getTypeDBBinary());
         command.add("cluster");
-        command.add("--address");
-        command.add(getAddressTripletString(ports));
-        peerPorts.forEach(peerPort -> {
-            command.add("--peer");
-            command.add(getAddressTripletString(peerPort));
-        });
-        command.add("--data");
-        command.add(dataDir.toAbsolutePath().toString());
+        serverOpts.forEach((key, value) -> command.add(key + "=" + value));
         return command;
     }
 
-    private String getAddressTripletString(Triple<Integer, Integer, Integer> ports) {
-        return host() + ":" + ports.first() + ":" + ports.second() + ":" + ports.third();
+    private static Map<String, String> portOptions(Ports ports) {
+        Map<String, String> options = new HashMap<>();
+        options.put(OPT_ADDR, host() + ":" + ports.external());
+        options.put(OPT_INTERNAL_ADDR_ZMQ, host() + ":" + ports.internalZMQ());
+        options.put(OPT_INTERNAL_ADDR_GRPC, host() + ":" + ports.internalGRPC());
+        return options;
+    }
+
+    private static Map<String, String> peerOptions(Set<Ports> peers) {
+        Map<String, String> options = new HashMap<>();
+        int index = 0;
+        for (Ports peer: peers) {
+            String addrKey = OPT_PEERS_ADDR.replace("{peer-id}", "peer-" + index);
+            String intAddrZMQKey = OPT_PEERS_INTERNAL_ADDR_ZMQ.replace("{peer-id}", "peer-" + index);
+            String intlAddrGRPCKey = OPT_PEERS_INTERNAL_ADDR_GRPC.replace("{peer-id}", "peer-" + index);
```

Comment:
so these can become `OPTS_PEERS_ADDR.format(index)`, where you move `peer-%d` inside of the template